diff --git a/Core/wbInterface.pas b/Core/wbInterface.pas
index b51ff527..6d73fb8d 100644
--- a/Core/wbInterface.pas
+++ b/Core/wbInterface.pas
@@ -2184,6 +2184,7 @@ type
     procedure SetTreeHead(aValue: Boolean);     // Make the element a "header record" in the tree navigator;
     function GetTreeBranch: Boolean;            // Is the element included in a "leaf" expected to be displayed in the view pane
     procedure SetTreeBranch(aValue: Boolean);   // Make the element included in a "leaf" visible in the tree navigator;
+    function GetTerminator: Boolean;
 
     procedure ToString(var Result : string; const aElement: IwbElement; aType: TwbCallbackType);
     function GetSummaryLinksTo(const aElement: IwbElement): IwbElement;
@@ -2208,6 +2209,8 @@ type
 
     property TreeBranch: Boolean
       read GetTreeBranch write SetTreeBranch;
+    property Terminator: Boolean
+      read GetTerminator;
   end;
 
   IwbSignatureDef = interface(IwbNamedDef)
@@ -2472,6 +2475,7 @@ type
   IwbSubRecordDef = interface(IwbRecordMemberDef)
     ['{D848E426-8768-45F4-B192-4DEFBE34D40A}']
     function GetValue: IwbValueDef;
+    function GetHasUnusedData: Boolean;
     procedure HasUnusedData;
 
     function IncludeFlagOnValue(aFlag: TwbDefFlag; aOnlyWhenTrue : Boolean = True): IwbSubRecordDef{Self};
@@ -2485,6 +2489,7 @@ type
     function SetLinksToCallbackOnValue(const aCallback: TwbLinksToCallback): IwbSubRecordDef{Self};
 
     property Value: IwbValueDef read GetValue;
+    property HasUnusedData: Boolean read GetHasUnusedData;
   end;
 
   IwbSubRecordWithStructDef = interface(IwbSubRecordDef)
@@ -2670,6 +2675,7 @@ type
     function GetSorted: Boolean;
     function GetCanAddTo: Boolean;
     function GetCountCallBack: TwbCountCallback;
+    function GetTerminated: Boolean;
     function GetPrefixSize(aBasePtr: Pointer): Integer;
     function GetPrefixLength(aBasePtr: Pointer): Integer;
     function GetPrefixCount(aBasePtr: Pointer): Cardinal;
@@ -2709,6 +2715,8 @@ type
 
     property CountCallBack: TwbCountCallback
       read GetCountCallback;
+    property Terminated: Boolean
+      read GetTerminated;
     property WronglyAssumedFixedSizePerElement: Integer
       read GetWronglyAssumedFixedSizePerElement;
 
@@ -5734,6 +5742,7 @@ type
     procedure SetTreeHead(aValue: Boolean);     // Make the element a "header record" in the tree navigator;
     function GetTreeBranch: Boolean;            // Is the element included in a "leaf" expected to be displayed in the view pane
     procedure SetTreeBranch(aValue: Boolean);   // Make the element included in a "leaf" visible in the tree navigator;
+    function GetTerminator: Boolean;
 
     procedure ToString(var Result : string; const aElement: IwbElement; aType: TwbCallbackType); reintroduce; virtual;
     function GetSummaryLinksTo(const aElement: IwbElement): IwbElement; virtual;
@@ -5986,6 +5995,7 @@ type
 
     {---IwbSubRecordDef---}
     function GetValue: IwbValueDef;
+    function GetHasUnusedData: Boolean;
     procedure HasUnusedData;
     function CanHandle(const aContainer     : IwbContainerElementRef;
                              aSignature     : TwbSignature;
@@ -6964,6 +6974,7 @@ type
     function GetSorted: Boolean;
     function GetCanAddTo: Boolean;
     function GetCountCallBack: TwbCountCallback;
+    function GetTerminated: Boolean;
     function GetPrefixSize(aBasePtr: Pointer): Integer;
     function GetPrefixLength(aBasePtr: Pointer): Integer;
     function GetPrefixCount(aBasePtr: Pointer): Cardinal;
@@ -10341,6 +10352,11 @@ begin
   Result := ndTreeHead;
 end;
 
+function TwbNamedDef.GetTerminator: Boolean;
+begin
+  Result := ndTerminator;
+end;
+
 procedure TwbNamedDef.InitFromParentBeforeChildren;
 begin
   if not (IsUnknown or ndUnused) and (ndName = '') then begin
@@ -11318,6 +11334,11 @@ begin
   Result := 'SubRecord of ' + GetValue.GetDefTypeName;
 end;
 
+function TwbSubRecordDef.GetHasUnusedData: Boolean;
+begin
+  Result := srHasUnusedData;
+end;
+
 function TwbSubRecordDef.GetValue: IwbValueDef;
 begin
   Result := srValue;
@@ -13514,6 +13535,11 @@ begin
   Result := arCountCallBack;
 end;
 
+function TwbArrayDef.GetTerminated: Boolean;
+begin
+  Result := arTerminated;
+end;
+
 function TwbArrayDef.GetCountPath: string;
 begin
   Result := arCountPath;
